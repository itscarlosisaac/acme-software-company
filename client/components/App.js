import React, { Component } from 'react'

// Importing components
import Header from './Header';
import Footer from './Footer';
import ImageCardList from './ImageCardList';
import LightBox from './Lightbox';
import EventEmitter from '../events/emitter'

export default class App extends Component {

  state = {
    loadedGif:[],
    count: 0,
    offset: 0,
    showLightBox: false,
    currentLightBoxImage: {},
    hasSearch: false
  }

  componentWillMount = () => {
    EventEmitter.on('LoadGifs', (data) => {
      let upcomingData = [ ...data ]
      let upcomingSearch = data.length > 1;
      this.setState({ 
        loadedGif: upcomingData,
        count: upcomingData.length,
        hasSearch: upcomingSearch
       })
    })
  }

  handleNext = () => {
    let currentIndex = this.state.loadedGif.indexOf(this.state.currentLightBoxImage);
    let nextImage = this.state.loadedGif[currentIndex + 1];
    if( nextImage !== undefined){
      this.setState({ currentLightBoxImage: nextImage })
    }
  }

  handlePrev = () => {
    let currentIndex = this.state.loadedGif.indexOf(this.state.currentLightBoxImage);
    let prevImage = this.state.loadedGif[currentIndex - 1];
    if( prevImage !== undefined){
      this.setState({ currentLightBoxImage: prevImage })
    }
  }

  handleThumbClick = (id) => {
    let curr = this.state.loadedGif.filter( (item) => item.id === id )
    this.setState({ 
      showLightBox: true,
      currentLightBoxImage: curr[0]
    })
  }

  closeLightBox = () => {
    this.setState({ showLightBox: false })
  }

  render() {
    return (
      <div className="app__container">
        <Header shrinked={this.state.hasSearch} />

        <ImageCardList 
          handleThumbClick={this.handleThumbClick}
          cards={this.state.loadedGif} />

        {
          this.state.showLightBox ? 
          <LightBox 
            close={this.closeLightBox}
            image={this.state.currentLightBoxImage} 
            prev={this.handlePrev}
            next={this.handleNext}  /> : ''
        }

        <Footer />
      </div>
    )
  }
}
