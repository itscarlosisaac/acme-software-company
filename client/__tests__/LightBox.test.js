import React from 'react';
import App from '../components/App';
import LightBox from '../components/LightBox';

import { configure, shallow } from 'enzyme';
import Adapter from  'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter() });
const image = {
  title: "My image",
  images: {
    downsized: {
      url: "https://avatars1.githubusercontent.com/u/19247005?s=60&v=4"
    }
  }
}
describe('<LoadMoreButton>', () => {
  it('Should render component', () => {
    const wrapper = shallow(<LightBox image={image} next={()=> {}} prev={()=>{}} close={()=>{}} />);
    expect(wrapper.find('.app__lightbox__close').length).toBe(1);
    expect(wrapper.find('.app__lightbox__prev').length).toBe(1);
    expect(wrapper.find('.app__lightbox__next').length).toBe(1);
    expect(wrapper.find('.app__lightbox__next').length).toBe(1);
    expect(wrapper.find('.app__lightbox__title').length).toBe(1);
    expect(wrapper.find('.app__lightbox__title').text().length).toBeGreaterThan(0)
  });

  it(`Should call next on click to next button`, () => {
    const mockCall = jest.fn();
    const wrapper = shallow(<LightBox image={image} next={mockCall} prev={()=>{}} close={()=>{}} />);
    wrapper.find('.app__lightbox__next').simulate('click')
    expect(mockCall.mock.calls.length).toEqual(1)
  });

  it(`Should call prev on click to next button`, () => {
    const mockCall = jest.fn();
    const wrapper = shallow(<LightBox image={image} next={()=>{}} prev={mockCall} close={()=>{}} />);
    wrapper.find('.app__lightbox__prev').simulate('click')
    expect(mockCall.mock.calls.length).toEqual(1)
  });

  it(`Should call close on click to next button`, () => {
    const mockCall = jest.fn();
    const wrapper = shallow(<LightBox image={image} next={()=>{}} prev={()=>{}} close={mockCall} />);
    wrapper.find('.app__lightbox__close').simulate('click')
    expect(mockCall.mock.calls.length).toEqual(1)
  });

  it(`Matches the snapshot`, () => {
    const tree = shallow(<LightBox image={image} next={()=> {}} prev={()=>{}} close={()=>{}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});