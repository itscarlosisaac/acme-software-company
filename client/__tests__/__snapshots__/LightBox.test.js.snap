// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMoreButton> Matches the snapshot 1`] = `
<div
  className="app__lightbox__container"
>
  <div
    className="app__lightbox__content"
  >
    <h2
      className="app__lightbox__close"
      onClick={[Function]}
    >
      Ã—
    </h2>
    <p
      className="app__lightbox__prev"
      onClick={[Function]}
    >
      Prev
    </p>
    <img
      alt="My image"
      src="https://avatars1.githubusercontent.com/u/19247005?s=60&v=4"
    />
    <h3
      className="app__lightbox__title"
    >
      My image
       
    </h3>
    <p
      className="app__lightbox__next"
      onClick={[Function]}
    >
      Next
    </p>
  </div>
</div>
`;
